builtin/logical/consul/path_token.go

```diff
@@ -103,13 +106,41 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 		})
 	}
 
+	aclServiceIdentities := []*api.ACLServiceIdentity{}
+	parsedServiceIdentities, err := parseSemicolonStringSlice(roleConfigData.ServiceIdentities)
```

Comment:
If we're going with a plugin-specific function that parses the service_identities string value, would it be worth to parse it all the way such that what gets returned is `[]*api.ACLNodeIdentity{}`? That we we avoid doing half of the parsing there and the other half down below.

Reply:
Good idea, I decided to move both node identity and service identity parsing out into functions because they're by far the bulkiest part of that code section.